src/batch/src/executor2/row_seq_scan.rs

```diff
@@ -48,15 +48,19 @@ impl<S: StateStore> RowSeqScanExecutor2<S> {
         stats: Arc<BatchMetrics>,
     ) -> Self {
         let schema = table.schema().clone();
-
+        // This is ugly but doing this can save a lot of trouble for unpinning the epoch associated
+        // with this iterator. We can treat the successful or failed creation of iterator just as
+        // the successful or failed scheduling of tasks. No extra communication path needs to be
+        // introduced to notify the batch query manager.
+        // let ret = table.iter(epoch).await.map_err(RwError::from);
```

Comment:
So error happens in the `execute` body cannot be propagated to the query manager properly? I'm still worried about using `block_on` in a sync function. 😢

Reply:
yes, I decided to change to async builder, block on will definitely block the entire runtime if there are too many row_seq_scan executor to be created and unlucky.